### PR TITLE
allow run in cluster

### PR DESCRIPTION
**Pull request by bborbe**

### CODE REVIEW

**Comment by bborbe** [comment]
Das logging koennen wir dann ja wieder umstellen. Aber die in cluster config brauchen wir schon.

**Comment by kwiesmueller** [comment]
kk

File: cmd/adeia/adeia.go [hunk 1]
```diff
@@ -70,9 +70,13 @@ func do() error {
 	if len(*namespacePtr) == 0 {
 		return errors.New("parameter namespace missing")
 	}
-	if len(*kubeconfigPtr) == 0 {
-		return errors.New("parameter kubeconfig missing")
-	}
+	glog.V(1).Infof("arg dry-run: %b", *dryRunPtr)
```
Comment (j-frost) [inline cmd/adeia/adeia.go]: This set of uniform statements looks a little whack to me -- is this idiomatic Go or something we could improve?

File: cmd/adeia/adeia.go [hunk 2]
```diff
@@ -91,6 +95,7 @@ func do() error {
 		},
 	}
 	if *dryRunPtr {
+		glog.V(2).Infof("dry-run=true => use print applier")
```
Comment (j-frost) [inline cmd/adeia/adeia.go]: This is not sparse logging for a machine to parse right? This is what people will probably see when they first execute the program. Do we want to formulate sentences here? Something like `param dry-run has value true => using print applier`?

File: ingress/k8s_applier.go [hunk 3]
```diff
@@ -21,13 +22,9 @@ type K8sApplier struct {
 
 // Apply a list of domains.
 func (a *K8sApplier) Apply(ingress *k8s_v1beta1.Ingress) error {
-	config, err := k8s_clientcmd.BuildConfigFromFlags("", a.Kubeconfig)
+	clientset, err := createClientSet(a.Kubeconfig)
```
Comment (j-frost) [inline ingress/k8s_applier.go]: k8s.io spell it `Clientset`, we probably should too.

File: ingress/k8s_applier.go [hunk 4]
```diff
@@ -45,3 +42,20 @@ func (a *K8sApplier) Apply(ingress *k8s_v1beta1.Ingress) error {
 	glog.V(0).Infof("ingress %s updated successful", ingress.Name)
 	return nil
 }
+
+func createClientSet(kubeconfig string) (k8s_kubernetes.Interface, error) {
```
Comment (j-frost) [inline ingress/k8s_applier.go]: We should probably rename this to `createClientset`; k8s.io call it `Clientset`.

File: ingress/k8s_applier.go (hunk 3 above)
Comment (j-frost) [reply ingress/k8s_applier.go]: Whoops, wrong place to comment this, see below.

File: ingress/k8s_applier.go [hunk 5]
```diff
@@ -45,3 +42,20 @@ func (a *K8sApplier) Apply(ingress *k8s_v1beta1.Ingress) error {
...
+	}
+	return k8s_kubernetes.NewForConfig(config)
+}
+
+func createConfig(kubeconfig string) (*k8s_rest.Config, error) {
+	if len(kubeconfig) > 0 {
+		glog.V(4).Infof("create kube config from flags")
+		return k8s_clientcmd.BuildConfigFromFlags("", kubeconfig)
+	}
+	glog.V(4).Infof("create kube config with in cluster config")
```
Comment (j-frost) [inline ingress/k8s_applier.go]: *within?

File: cmd/adeia/adeia.go (hunk 1 above)
Comment (bborbe) [reply cmd/adeia/adeia.go]: don't think so

**Pull request merged**